Add `javac` on WebAssembly demo. (#345)
graalvm/graalvm-demos: native-image/wasm-javac/README.md

# Compile `javac` to a Wasm Module

> In this example, we'll compile the `javac` tool to Wasm modules that can then run on the commandline or the browser.

Reviewer: ```suggestion
This demo illustrates how to use the WebAssembly backend of GraalVM to compile the `javac` tool into a Wasm module that can then run on the command-line or in the browser.
```

## Prerequisites

> These examples require:

Reviewer: ```suggestion
This demo requires:
```

> 1. An [Oracle GraalVM Early Access Build](https://github.com/graalvm/oracle-graalvm-ea-builds), version 25.0.0-.ea.15 or later.

Reviewer: ```suggestion
1. An [Early Access Build](https://github.com/graalvm/oracle-graalvm-ea-builds) of Oracle GraalVM for JDK 25 or later.
```

    For example, using SDKMAN!: `sdk install java 25.ea.15-graal`

> 2. The [Binaryen toolchain](https://github.com/WebAssembly/binaryen) on the sytem path, version 119 or later

Reviewer: ```suggestion
2. The [Binaryen toolchain](https://github.com/WebAssembly/binaryen) in version 119 or later and on the system path.
```

    For example, using Homebrew: `brew install binaryen`

> ## Run `javac` CLI

Reviewer: ```suggestion
## Run `javac` on Node
```

> 

Reviewer 2: Double empty line

> 1. Build the Wasm binary:

Reviewer: ```suggestion
1. Build the Wasm module:
```

    ```bash

>     mvn -Pnative package

Author: Added the maven wrapper

    ```
    Uses the [Native Build Tools](https://graalvm.github.io/native-build-tools/latest/index.html) to run the `native-image` executable from Maven.

>     This will generate the Wasm file and its JavaScript wrapper in the `target` directory.

Reviewer: ```suggestion
    This command generates a Wasm file and a corresponding JavaScript binding in the `target` directory.
```

2. Compile a Java file:
    ```bash
    node target/javac.js HelloWorld.java
    ```
    The resulting `HelloWorld.class` file is placed next to `HelloWorld.java`.
    This requires Node.js 22 or later.

> ## Run Web Demo

Reviewer: I would put this part of the demo first.

```suggestion
## Run `javac` in the browser
```

> 1. Build the Wasm binary:

Reviewer: ```suggestion
1. Build the Wasm module:
```

    ```bash
    mvn -Pjavacweb package
    ```

>     This generates the Wasm and JS files into the `web` directory.

Reviewer: ```suggestion
    This command generates a Wasm file and a corresponding JavaScript binding in the `web` directory.
```

2. Run a local web server in the `web` directory:
    ```bash
    cd web
    jwebserver
    ```
    This will serve the `web` directory locally on port `8000`.

> 3. Navigate to http://localhost:8000 to compile Java programs in the browser:

Reviewer 2: Can we add some short summary for this demo  at the end? Or some links to read more?
